homeassistant/components/minecraft_server/__init__.py

```diff
@@ -23,8 +30,20 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Minecraft Server from a config entry."""
 
+    # Migrate server type if required.
+    if CONF_TYPE not in entry.data:
+        await _async_migrate_config_entry_data_key_type(hass, entry)
```

Comment:
```suggestion
```

Please use the `_migrate_entity_unique_id` (as you used before) for migration **(recommended)** or use `entry.data.get(CONF_TYPE, MinecraftServerType.JAVA_EDITION)`

Reply:
Did you mean `async_migrate_entry` instead of `_migrate_entity_unique_id`? If so you're suggesting to revert [commit 1a6009f4c5b2a66656399e333de38ed51bdf3508](https://github.com/home-assistant/core/pull/100925/commits/1a6009f4c5b2a66656399e333de38ed51bdf3508) and increment the config flow version again to 4?

Reply:
Quickly discussed that internally and you should use `entry.data.get(CONF_TYPE, MinecraftServerType.JAVA_EDITION)` instead.
Sorry for suggesting the migration